Add context around choosing SQL and Warehouse SQL cells (#1950)
astronomer/docs: astro/cloud-ide/run-sql.md

description: Learn how to run SQL queries by creating and configuring SQL cells in the Astro Cloud IDE.
---

A SQL cell contains a SQL query that you can run in isolation or as a dependency in your pipeline. Create SQL cells and Warehouse SQL cells to execute SQL as part of your data pipeline. 

## Prerequisites 

- An IDE project and pipeline. See [Step 2: Create a pipeline](/astro/cloud-ide/quickstart.md#step-2-create-a-pipeline).
- A database connection. See [Step 5: Create a database connection](/astro/cloud-ide/quickstart.md#step-5-create-a-database-connection).
- For Warehouse SQL cells, you need write permissions to the data warehouse in your database connection. 

## Choose a SQL cell type

> A SQL cell runs a SQL query on a database connection and stores the results of the query in an XCom file for use in other cells. A Warehouse SQL cell runs a SQL query on an external database and stores the results in your data warehouse.

Author: ```suggestion
A SQL cell runs a SQL query on a database connection and stores the results of the query in an XCom value for use in other cells. A Warehouse SQL cell runs a SQL query on an external database and stores the results in your data warehouse.
```

> Both SQL and Warehouse SQL cells run SQL queries. A SQL cell runs a SQL query on a database connection and stores the results of the query in an XCom file for use in other cells. A Warehouse SQL cell runs a SQL query on an external database and stores the results in your data warehouse.

Reviewer: ```suggestion
Both SQL and Warehouse SQL cells run SQL queries. A SQL cell runs a SQL query on a database connection and stores the results of the query in an XCom value for use in other cells. A Warehouse SQL cell runs a SQL query on an external database and stores the results in your data warehouse.
```

> A SQL cell runs a SQL query on a database connection and stores the results of the query in an XCom value for use in other cells. A Warehouse SQL cell runs a SQL query on an external database and stores the results in your data warehouse. For more information about how these cells work, see [How SQL cells work](#how-sql-cells-work).

Author: ```suggestion
A SQL cell runs a SQL query against a database connection and stores the results of the query in an XCom value for use in other cells. A Warehouse SQL cell runs a SQL query against a database connection and stores the results in your data warehouse. For more information about how these cells work, see [How SQL cells work](#how-sql-cells-work).
```

Both cell types execute SQL queries, but there are some scenarios where the use of one cell type is preferable. The following table lists the scenarios where Astronomer recommends using a specific type of SQL cell.
  
| Scenario                                                                                            | SQL cell | Warehouse SQL cell |
| --------------------------------------------------------------------------------------------------- | -------- | ------------------ |
| I don't have write access to an external database.                                                  | ✔️        |                    |
| I'm going to use the output of the query outside of my external database, such as in a Python cell. | ✔️        |                    |
| I'm going to use the output of the query only in my external database.                              |          | ✔️                  |

> | I'm querying a large amount of data.                                                                |          | ✔️                  |

Reviewer: this may just be me but I don't like how the check mark isn't center aligned - is there anything we can do about that? maybe we switch this to two columns: scenario and cell type, where cell type is either SQL or Warehouse SQL?

Reviewer 2: You can add a colon on both sides of the dashes `:----:` to center align text in a table.

## Create a SQL cell

> When you can use two different types of cells to query SQL: SQL and Warehouse SQL cells. A SQL cell runs a SQL query against a provided database connection and stores the results of the query in an XCom for use in other cells. A Warehouse SQL cell runs a SQL query against a provided external database and stores the results in your data warehouse.

Reviewer 2: ```suggestion
You can use SQL and Warehouse SQL cells to query SQL. A SQL cell runs a SQL query against a provided database connection and stores the results of the query in an XCOM file for use in other cells. A Warehouse SQL cell runs a SQL query against a provided external database and stores the results in your data warehouse.
```

> Both cell types execute SQL queries, but there are some scenarios where you should use one cell type over the other. The following table contains all scenarios where Astronomer recommends using a specific type of SQL cell.

Reviewer 2: ```suggestion
Both cell types execute SQL queries, but there are some scenarios where the use of one cell type is preferable. The following table lists the scenarios where the use of a specific type of SQL cell is recommended.
```

  
| Scenario                                                                                             | SQL cell | Warehouse SQL cell |
| ---------------------------------------------------------------------------------------------------- | -------- | ------------------ |
| I don't have write access to an external database.                                                   | ✔️        |                    |

> | I am going to use the output of the query outside of my external database, such as in a Python cell. | ✔️        |                    |

Reviewer 2: ```suggestion
| I'm going to use the output of the query outside of my external database, such as in a Python cell. | ✔️        |                    |
```
You used contractions previously, so you should continue using them to keep the list parallel.

> | I am going to use the output of the query only in my external database.                              |          | ✔️                  |

Reviewer 2: ```suggestion
| I'm going to use the output of the query only in my external database.                              |          | ✔️                  |
```

> | I am querying a large amount of data.                                                                |          | ✔️                  |

Reviewer 2: ```suggestion
| I'm querying a large amount of data.                                                                |          | ✔️                  |
```

> Regardless of which cell type you choose, you can use the results of your queries in downstream cells. However, using the output of a Warehouse cell in a downstream cell requires fetching the output from your external database, which can take longer than fetching the output of a SQL cell from XComs.

Reviewer 2: ```suggestion
Regardless of the cell type you choose, you can use your query results in downstream cells. However, using the output of a Warehouse cell in a downstream cell requires fetching the output from your external database, which can take longer than fetching the output of a SQL cell with XCOM.
```
